MaxMind.Db.Benchmark/Program.cs

```diff
@@ -1,98 +1,212 @@
-﻿#region
-
+﻿using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Configs;
+using BenchmarkDotNet.Running;
+using MaxMind.Db;
 using System;
 using System.Collections.Generic;
-using System.Diagnostics;
-using System.IO;
+using System.Collections.ObjectModel;
+using System.Linq;
 using System.Net;
 
-#endregion
+BenchmarkRunner.Run<CityBenchmark>(new DebugInProcessConfig());
 
-namespace MaxMind.Db.Benchmark
+[MemoryDiagnoser]
+public class CityBenchmark
 {
-    public class Country
+    // A random IP that has city info.
+    private static readonly IPAddress IpAddress = IPAddress.Parse("214.0.0.0");
```

Comment:
I'd prefer if we had a larger collection of IP addresses. I can see how a single IP with all the relevant info would be useful when testing deserialization, but it may be misleading in other context, e.g., performance of the tree traversal. It also may be misleading for deserialization if we ever add any sort of caching.

Perhaps we could use an array of random IPs generated with a fixed seed during setup. We could have an option to specify a custom list when you really want to focus in on a few examples.

Reply:
I picked this IP because it seems like the only one with city info. Maybe I could extra benchmark methods for different use cases. Could be:
- Empty: an empty response object to measure the tree traversal
- Country: an IP that has country info

Wdyt?

Reply:
I still think a large number of IPs would provide a more reliable benchmark. The tree will vary significantly between networks, even if the data for the network in the data section is identical. Also, the IPs that have a full city record, a country-only record, or are missing is likely to change over time. If we generated a list of IPs during start-up using a fixed random seed, that should provide repeatability. For your particular use case, I think it may be useful to be able to override that with a list of IPs, e.g., in a text file, flag, or environment variable.

Reply:
Done